nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -1894,14 +1899,47 @@ public Environment getEnv() {
         return env;
     }
 
+    /**
+     * @since 10.2
+     * @param propsFile Properties file
+     * @return String with the charset encoding for this file
+     */
+    public static String checkFileCharset(File propsFile) throws IOException {
+        String[] charsetsToBeTested = {"US-ASCII", "UTF-8", "ISO-8859-1"};
+        for (String charsetName : charsetsToBeTested) {
+            Charset charsetDet = Charset.forName(charsetName);
+            CharsetDecoder decoder = charsetDet.newDecoder();
+            decoder.reset();
+
+            BufferedInputStream input = new BufferedInputStream(new FileInputStream(propsFile));
```

Comment:
Do we need to close it ?

Reply:
yes, "try with resources" would be good.

Reply:
```java
try(InputStream fileInputStream = Files.newInputStream(propsFile.toPath();
      BufferedInputStream input = new BufferedInputStream(fileInputStream)
) {
}
```